commcare_connect/opportunity/visit_import.py

```diff
@@ -290,14 +305,24 @@ def get_exchange_rate(currency_code, date=None):
     if currency_code == "USD":
         return 1
 
-    base_url = "https://openexchangerates.org/api"
-    if date:
-        url = f"{base_url}/historical/{date.strftime('%Y-%m-%d')}.json"
-    else:
-        url = f"{base_url}/latest.json"
-    url = f"{url}?app_id={settings.OPEN_EXCHANGE_RATES_API_ID}"
-    rates = json.load(urllib.request.urlopen(url))
-    return rates["rates"].get(currency_code)
+    today = now().date()
+    rate = (
+        ExchangeRate.objects.filter(currency_code=currency_code, rate_date=date or today)
```

Comment:
nit, `date or today` would be a little cleaner outside the arguments (like the line above)